tests/xdc/base.go

```diff
@@ -100,6 +101,8 @@ func (s *xdcBaseSuite) setupSuite(clusterNames []string, opts ...tests.Option) {
 	if s.dynamicConfigOverrides == nil {
 		s.dynamicConfigOverrides = make(map[dynamicconfig.Key]interface{})
 	}
+	// Always force SA cache readthrough for tests to avoid propagation delays causing flakiness.
+	s.dynamicConfigOverrides[dynamicconfig.ForceSearchAttributesCacheRefreshOnRead.Key()] = true
```

Comment:
Should we just put this here so it applies to all tests?
https://github.com/temporalio/temporal/blob/main/tests/onebox.go#L194
(and remove [this one](https://github.com/temporalio/temporal/blob/main/tests/functional_test_base.go#L130
))